src/OpenCV/Core/Types/Mat.hs

```diff
@@ -242,3 +262,87 @@ matCopyToM dstM (V2 x y) src mbSrcMask = ExceptT $
           --                       )
           --                 );
           -- srcPtr->copyTo(dstRoi);
+
+
+class All (p :: k -> Constraint) (xs :: [k])
+instance All p '[]
+instance (p x, All p xs) => All p (x ': xs)
+
+class IsStatic (ds :: DS a)
```

Comment:
We also need an unexported `Private`/`Internal` super class constraint to ensure users can't create their own `IsStatic` instances.